src/admin/components/views/collections/Edit/index.tsx

```diff
@@ -59,24 +77,25 @@ const EditView: React.FC<IndexProps> = (props) => {
     if (!isEditing) {
       setRedirect(`${admin}/collections/${collection.slug}/${json?.doc?.id}`);
     } else {
-      const preferences = await getDocPreferences();
-      const state = await buildStateFromSchema({ fieldSchema: collection.fields, preferences, data: json.doc, user, id, operation: 'update', locale, t });
-      setInternalState(state);
+      buildState(json.doc, {
+        fieldSchema: collection.fields,
+      });
     }
-  }, [admin, collection.fields, collection.slug, getDocPreferences, getDocPermissions, getVersions, id, isEditing, locale, t, user]);
```

Comment:
Could this have been kept the same except for removing the `user` dependency?

Reply:
Yea it could have, but this change standardizes the duplicative logic between this `useCallback` and `useEffect`.